core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -92,28 +97,30 @@ object StaticFile {
         val expired = (ifModifiedSince, lastmod).mapN(_.date < _).getOrElse(true)
 
         if (expired) {
-          val len = urlConn.getContentLengthLong
-          val headers = Headers(
-            lastmod.map(`Last-Modified`(_)),
-            nameToContentType(url.getPath),
-            if (len >= 0) `Content-Length`.unsafeFromLong(len)
-            else `Transfer-Encoding`(TransferCoding.chunked.pure[NonEmptyList])
-          )
-
-          F.blocking(urlConn.getInputStream)
-            .redeem(
-              recover = {
-                case _: FileNotFoundException => None
-                case other => throw other
-              },
-              f = { inputStream =>
-                Some(
-                  Response(
-                    headers = headers,
-                    body = readInputStream[F](F.pure(inputStream), DefaultBufferSize)
-                  ))
-              }
+          etagCalculator(file).map(ETag(_)).flatMap { etag =>
```

Comment:
So if `file` is suspect (see comment above), then the `etagCalculator` that takes a `File` is not the right tool for this method.  The etag calculator in this method should probably be `URL => F[ETag]`.  A default implementation could get lastmodified and length from the `urlConn` (examples are already in this function) and then operate similarly to the file-based `calcETag`.  That default implementation should also have a `Sync` constraint on `F`... which will also complicate passing it as a default argument.

I'll leave that as an exercise (it's late here), but if that doesn't make sense, don't spin your wheels too hard on it: someone will be awake soon to give more suggestions. :smile:

Reply:
Changing it to default arguments would be for Issue #2072 right?  
Should I change ``StaticFile`` to an algebra as well, and then add in default parameters rather than overloading like I just did?